protocols/generic/generic.go

```diff
@@ -65,29 +71,54 @@ func build(tn *testnet.TestNet) error {
 
 	tn.BuildState.IncrementBuildProgress()
 
-	err := helpers.AllNewNodeExecCon(tn, func(client ssh.Client, _ *db.Server, node ssh.Node) error {
-		files, ok := tn.LDD.Params["files"].([]interface{})
-		if !ok {
-			err := fmt.Errorf("tn.LDD.Params['files'] is not a map[string]string, it is a %v", reflect.TypeOf(tn.LDD.Params["files"]).String())
+	topology, err := getTopology(tn)
+	if err != nil {
+		return util.LogError(err)
+	}
+
+	return buildNetwork(tn, nodeKeyPairs, topology)
+}
+
+func idString(k crypto.PrivKey) string {
+	pid, err := peer.IDFromPrivateKey(k)
+	if err != nil {
+		panic(err) //TODO why tf panic
+	}
+	return pid.Pretty()
+}
+
+func getTopology(tn *testnet.TestNet) (topology, error) {
+	networkTopology := fmt.Sprintf("%v", tn.LDD.Params["network-topology"])
+
+	switch networkTopology {
+	case "all":
+		return all, nil
+	case "sequence":
+		return sequence, nil
+	case "randomTwo":
+		return randomTwo, nil
+	default:
+		return all, util.LogError(fmt.Errorf("unsupported network topology, %v", networkTopology))
+	}
+}
+
+func buildNetwork(tn *testnet.TestNet, nodeKeyPairs map[string]crypto.PrivKey, networkTopology topology) error {
+	return helpers.AllNewNodeExecCon(tn, func(client ssh.Client, _ *db.Server, node ssh.Node) error {
+		err := copyFiles(tn, client, node)
+		if err != nil {
 			return util.LogError(err)
 		}
 
-		filesToCopy := files[node.GetRelativeNumber()]
+		params, err := createDefaultParams(tn, node)
 
-		fileMap, ok := filesToCopy.(map[string]interface{})
-		if !ok {
-			err := fmt.Errorf("filesToCopy is a %v", reflect.TypeOf(filesToCopy).String())
+		peers, err := createPeers(tn, node, nodeKeyPairs, networkTopology)
```

Comment:
`createPeers(...)` will need to be done in a loop and fed the correct peers to add into a file